cmd/storage-errors.go

```diff
@@ -157,6 +162,7 @@ func osErrToFileErr(err error) error {
 		return errFaultyDisk
 	}
 	if isSysErrInvalidArg(err) {
+		logger.LogIf(context.Background(), err)
```

Comment:
We don't really do logging here, and it seems like an unintentional side effect. 

You could argue that a lot of these would warrant a log entry `isSysErrHandleInvalid, isSysErrIO, isSysErrTooManyFiles,isSysErrNoSpace` seems pretty severe, but aren't logged.

I guess it is being ignored since it returns a "harmless" error. Maybe we should "upgrade" it instead, maybe `errFaultyDisk`?

Reply:
yes, the only true reason for logging this is that we convert this error to errFileNotFound, so debugging this will be hard.

Reply:
The reason this will happen @vadmeste is very rare and we don't have to log this error. 

This is mainly a filesystem related error due to some flags not implemented etc.

EINVAL to errFileNotFound was added because some random characters in object name with O_DIRECT throw EINVAL and the object doesn't exist..

Reply:
We actually need to log this because it is very rare. My only aim of this is that it will be easier for us to debug when this happens. If an S3 application complains about a 404 error, we can waste multiple hours debuggining to understand why this happened.